redis_rate_limit/__init__.py

```diff
@@ -37,29 +38,82 @@ class TooManyRequests(Exception):
     pass
 
 
-class RateLimit(object):
+class RateLimiter(object):
     """
-    This class offers an abstraction of a Rate Limit algorithm implemented on
-    top of Redis >= 2.6.0.
+    Rate limit factory.
     """
-    def __init__(self, resource, client, max_requests, expire=None):
+    GLOBAL_PREFIX = 'rate_limit:'
+
+    def __init__(self, resource, max_requests, expire=None, redis_pool=REDIS_POOL):
         """
-        Class initialization method checks if the Rate Limit algorithm is
+        Class initialization method lazily checks if the Rate Limit algorithm is
         actually supported by the installed Redis version and sets some
         useful properties.
+        :param resource: resource identifier string (i.e. ‘user_pictures’)
+        :param max_requests: integer (i.e. ‘10’)
+        :param expire: seconds to wait before resetting counters (i.e. ‘60’)
+        :param redis_pool: instance of redis.ConnectionPool.
+               Default: ConnectionPool(host='127.0.0.1', port=6379, db=0)
+        """
+        self._redis_pool = redis_pool
+        self._redis = lazy_object_proxy.Proxy(self._get_redis)
+        self._prefix = "{0}{1}".format(self.GLOBAL_PREFIX, resource)
+        self._max_requests = max_requests
+        self._expire = expire or 1
+
+    def limit(self, client=''):
+        return RateLimit(
+            redis=self._redis,
+            key='{0}_{1}'.format(self._prefix, client),
+            max_requests=self._max_requests,
+            expire=self._expire)
+
+    def reset(self):
+        """
+        Deletes all keys for this resource.
+        """
+        for rate_limit_key in self._redis.keys('{0}*'.format(self._prefix)):
```

Comment:
we can use scan here instead of keys, keys blocks Redis from doing other operations.